TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/workflow/models/secret_store.py

```diff
@@ -0,0 +1,354 @@
+"""Storage implementation for sensitive credentials using Kubernetes Secrets."""
+
+import logging
+from typing import Optional, List, Dict, Set
+
+from kubernetes import client, config
+from kubernetes.client.rest import ApiException
+
+logger = logging.getLogger(__name__)
+
+
+class SecretStore:
+    """
+    Secret store using Kubernetes Secrets.
+    Stores sensitive credentials (usernames, passwords, tokens) persistently in the cluster.
+
+    See WorkflowConfigStore, which inspired this class
+    """
+
+    def __init__(
+            self,
+            namespace: str = "default",
+            secret_prefix: str = "workflow-secret",
+            default_labels: Optional[Dict[str, str]] = None,
+            k8s_client: Optional[client.CoreV1Api] = None
+    ):
+        """Initialize the store with Kubernetes configuration
+
+        Args:
+            namespace: Kubernetes namespace to use for Secrets
+            secret_prefix: Prefix for Secret names (unused but kept for consistency)
+            default_labels: Default labels to apply to all secrets created by this store
+            k8s_client: Optional pre-configured Kubernetes client (for testing)
+        """
+        self.namespace = namespace
+        self.secret_prefix = secret_prefix
+
+        # Set default labels - can be overridden per secret
+        self.default_labels = default_labels or {
+            "app": "migration-assistant",
+            "component": "workflow-secret",
+        }
+
+        if k8s_client:
+            # Use provided client (useful for testing)
+            self.v1 = k8s_client
+            logger.info("Using provided Kubernetes client")
+        else:
+            # Load Kubernetes configuration
+            try:
+                # Try to load in-cluster config first (when running in a pod)
+                config.load_incluster_config()
+                logger.info("Loaded in-cluster Kubernetes configuration")
+            except config.ConfigException:
+                try:
+                    # Fall back to local kubeconfig (for development/minikube)
+                    config.load_kube_config()
+                    logger.info("Loaded local Kubernetes configuration")
+                except config.ConfigException as e:
+                    logger.error(f"Failed to load Kubernetes configuration: {e}")
+                    raise
```

Comment:
my PR adds a `KubernetesConfigNotFoundError` to `workflow.models.utils` and we should use that here as well.